src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs

```diff
@@ -263,6 +335,64 @@ private static string GetAbsolutePath(string path, string baseDirectoryPath)
             return projectResult;
         }
 
+        /// <summary>
+        /// Reload the given projects based on their contents on disk.
+        /// </summary>
+        /// <param name="projectIds">The <see cref="ProjectId" /> set to be reloaded</param>
+        /// <param name="targets">The set of targets to execute on all projects in the solution</param>
+        /// <param name="msbuildLogger">An optional <see cref="ILogger"/> that will log msbuild results..</param>
+        /// <param name="progress">An optional <see cref="IProgress{T}"/> that will receive updates as the project is opened.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> to allow cancellation of this operation.</param>
+        public async Task<ImmutableArray<Project>> ReloadProjectsAsync(
+            ImmutableArray<ProjectId> projectIds,
```

Comment:
Should this API take a list of file paths? It seems we're going from project id -> file path, but ultimately any caller would be starting with a file path and have to find IDs, right to consider multi-targeting cases?